google/cloud/storage/blob.py

```diff
@@ -829,9 +829,12 @@ def _get_download_url(
         """
         name_value_pairs = []
         if self.media_link is None:
-            base_url = _DOWNLOAD_URL_TEMPLATE.format(
-                hostname=client._connection.API_BASE_URL, path=self.path
+            hostname = (
+                client._connection.API_BASE_URL
+                if not hasattr(client._connection, "get_api_base_url_for_mtls")
```

Comment:
Is this something we can also simplify by requiring google-cloud-core 1.6.0 in setup.py? (also same idiom below)

Reply:
cannot upgrade it yet as stated above

Reply:
Could you introduce a single helper internal helper method to reduce duplication of this check and clarify that it should be cleaned up once version 1.6.0 is stable?

Reply:
done